### PR TITLE
#3855: turned error to warning

### PR DESCRIPTION
**Pull request by ppasler**
#3855: removed the KeyError and added a warning instead

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Look at the AppVeyor log -- you need to modify the corresponding test. You can use `with warnings.catch_warnings(record=True) as w:`, see how we do it elsewhere.

**Comment by ppasler** [comment]
Darn, I've tested it against the installed master...

File: mne/io/tests/test_meas_info.py [hunk 1]
```diff
@@ -317,7 +319,9 @@ def test_check_consistency():
 
     info2 = info.copy()
     info2['filename'] = 'foo'
-    assert_raises(KeyError, info2._check_consistency)
+    with warnings.catch_warnings(record=True) as w:
+        info2._check_consistency()
+    assert len(w) == 1
```
Comment (Eric89GXL) [inline mne/io/tests/test_meas_info.py]: do not use `assert` in tests, use `assert_equal` here or even better:
```
assert_equal(len(w), 1)
assert_true(all('filename' in str(ww.message) for ww in w))
```

Comment (ppasler) [inline mne/io/tests/test_meas_info.py]: copy/paste... I fixed it in the other classes too.

File: mne/beamformer/tests/test_dics.py [hunk 2]
```diff
@@ -220,7 +220,7 @@ def test_dics_source_power():
     noise_csds = data_csds
     with warnings.catch_warnings(record=True) as w:
         dics_source_power(epochs.info, forward, noise_csds, data_csds)
-    assert len(w) == 1
+        assert_equal(len(w), 1)
```
Comment (Eric89GXL) [inline mne/beamformer/tests/test_dics.py]:
```
NameError: global name 'assert_equal' is not defined
```
Also, you don't need to indent this (no need to be within the warning-catching context anymore).

**Comment by Eric89GXL** [comment]
Thanks @ppasler

**Comment by ppasler** [comment]
@Eric89GXL no problem! 
Is there a way to test against my local changes, even if I have another version of mne installed?

**Comment by Eric89GXL** [comment]
Running `nosetests ...` from within the mne-python git root should do it, because the `import mne` order should look in the current directory `mne-python/` first, which has the `mne` subdirectory in it.

**Comment by ppasler** [comment]
@Eric89GXL this works, thanks!
`nosetests mne\io\tests\test_meas_info.py`

However, running the test with normal Python, does not
`python mne\io\tests\test_meas_info.py`

_Is there a way to run all tests inside the mne dir?_

**Comment by Eric89GXL** [comment]
`nosetests` by itself should work

**Comment by ppasler** [comment]
Thanks you sir, this works like a charme!

**Pull request merged**